### ISSUE TITLE
Idea for a more generic ViewTarget

### ISSUE THREAD

**Issue by gergely-sallai** [coil-kt/coil#338]
**Is your feature request related to a problem? Please describe.**
I often use Coil to load images as background of a View, for which I had to duplicate code from `ImageViewTarget`. This could be eliminated if code could be re-used from said class.

**Describe the solution you'd like**
My proposed solution lifts out logic from `ImageViewTarget` into a more generic `GenericViewTarget` which can be extended to provide support for both ImageViews and View background image setting.

**Additional context**
I created a Gist for this so you can take a quick look, I'll gladly make a PR if needed.
https://gist.github.com/gergely-sallai/7b4a962725008575d8bffac5ea5285d0

**Comment by colinrtwhite** [comment]
Thanks for the suggestion. I'm a little apprehensive to add something like `GenericViewTarget` to Coil's API as it starts to create a deep inheritance tree which goes against the principle of [favouring composition over inheritance](https://en.wikipedia.org/wiki/Composition_over_inheritance). I think Glide has run into this problem over time as it has `BaseTarget`, `CustomTarget`, `CustomViewTarget`, `SimpleTarget`, and `ViewTarget` all which lift out some level of functionality.

Also if we introduce `GenericViewTarget` should we also introduce a `GenericViewTarget` variant which implements `ViewTarget` instead of `PoolableViewTarget`?

**Comment by gergely-sallai** [comment]
Yeah, this totally makes sense. I'll try to think how could the logic in `ImageViewTarget` be included in a component to be reusable and avoid inheritance.

What's your opinion about having `GenericViewTarget` (or similar) not on the API but internally and exposing only `ImageViewTarget` and `ViewBackgroundTarget` on the API? If someone has custom needs they can still do this without mandatory inheritance and reuse the basic components that are available on the API right now?
